Document/0x06a-Platform-Overview.md

```diff
@@ -78,29 +78,24 @@ Building encryption into the physical architecture makes it a default security c
 
-
 #### General Exploit Mitigations
 
 iOS currently implements two specific security mechanisms, namely address space layout randomization (ASLR) and eXecute Never (XN) bit, to prevent code execution attacks.
 
-ASLR is a technique that does the job of randomizing the memory location of the program executable, data, heap and stack on every execution of the program. As the shared libraries need to be static in order to be shared by multiple processes, the addresses of shared libraries are randomized every time the OS boots instead of every time when the program is invoked.
-
-Thus, this makes the specific memory addresses of functions and libraries hard to predict, thereby preventing attacks such as a return-to-libc attack, which relies upon knowing the memory addresses of basic libc functions.
+ASLR is a technique that does the job of randomizing the memory location of the program executable, data, heap and stack on every execution of the program. As the shared libraries need to be static in order to be shared by multiple processes, the addresses of shared libraries are randomized every time the OS boots instead of every time when the program is invoked. Thus, this makes specific memory addresses of functions and libraries hard to predict, thereby preventing attacks such as a return-to-libc attack, which relies upon the memory addresses of basic libc functions.
 
 <!-- TODO [Further develop section on iOS General Exploit Mitigation] -->
-
+XN mechanism allows iOS to mark certain memory segments and non-executable on a program’s stack and heap by default. In case of attack this will prevent malicious code inserted onto the stack or heap from execution.
 
 ### Software Development on iOS
 
-As with other platforms, Apple provides a Software Development Kit (SDK) for iOS that helps developers to develop, install, run and test native iOS Apps by offering different tools and interfaces. Xcode is an Integrated Development Environment (IDE) used for this purpose and iOS applications are implemented either by using Objective-C or Swift.
+Like other platforms, Apple provides a Software Development Kit (SDK) for iOS that helps developers to develop, install, run and test native iOS Apps by offering different tools and interfaces. Xcode is an Integrated Development Environment (IDE) used for this purpose. iOS applications are developed either by using Objective-C or Swift.

```

Comment:
I'd only insist on "Like" instead of "As with". All other changes are made for readability purpose only